pkg/payload/payload.go

```diff
@@ -308,19 +308,20 @@ func loadPayloadMetadata(releaseDir, releaseImage string) (*Update, error) {
 	if err != nil {
 		return nil, err
 	}
-	release.Image = releaseImage
-
-	imageRef, err := loadImageReferences(releaseDir)
-	if err != nil {
-		return nil, err
-	}
 
 	arch := string(release.Architecture)
 	if arch == "" {
 		arch = runtime.GOARCH
 		klog.V(2).Infof("Architecture from %s (%s) retrieved from runtime: %q", cincinnatiJSONFile, release.Version, arch)
 	}
 
+	release.Image = releaseImage
+
+	imageRef, err := loadImageReferences(releaseDir)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
This move only addresses a review comment from an earlier PR: https://github.com/openshift/cluster-version-operator/pull/1185#discussion_r2078497343